hack/install-vm.sh

```diff
@@ -141,17 +150,246 @@ EOF
+        rm -rf ${docker_gpg_key}
+        cmd="curl -fsSL ${docker_download_url}/gpg | gpg --dearmor -o ${docker_gpg_key}"
+        execute_command "$cmd" 1m
+        # Setup a stable repository
+        echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=${docker_gpg_key}] ${docker_download_url} \
+        $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        cmd="apt-get install -y docker-ce docker-ce-cli containerd.io"
+        execute_command "$cmd" 2m
+        ;;
+    esac
+}
+
+# Install all pre-requisite packages needed for install
+install_docker_prerequisite_packages() {
+    set +eo pipefail
+    # Install docker, docker-compose only when running services in containers
+    if [ "$DOCKERIZE" = true ]; then
+        docker version
+        if [[ "$?" -ne 0 ]]; then
+            echo "Docker is not installed"
+            install_container_runtime
+        fi
+        # Install docker-compose
+        docker-compose version
+        if [ "$?" -ne 0 ]; then
+            cmd="curl -L "${DOCKER_COMPOSE}-$(uname -s)-$(uname -m)" -o ${DOCKER_COMPOSE_DIR}"
+            execute_command "$cmd" 1m
+            if [ "$?" -eq 0 ]; then
+                echo "docker-compose is downloaded successfully"
+                chmod +x ${DOCKER_COMPOSE_DIR}
+                ln -s ${DOCKER_COMPOSE_DIR} /usr/bin/docker-compose
+            else
+                echoerr "docker-compose couldn't be downloaded, exiting"
+                exit 2
+            fi
+        fi
+    fi
+    set -eo pipefail
+}
+
+# Create docker-compose file with ovs and antrea-agent container specific parameters
+create_docker_compose_file() {
+    cat << EOF > $AGENT_CONF_PATH/docker-compose.yaml
+version: "3"
+
+services:
+  ovs:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: ovs
+    volumes:
+        - /var/log/openvswitch:/var/log/openvswitch
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /lib/modules:/lib/modules
+    command: $OVS_BIN
+    network_mode: host
+    cap_add:
+        - NET_ADMIN
+        - SYS_NICE
+        - SYS_ADMIN
+        - IPC_LOCK
+
+  antrea-agent:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: antrea-agent
+    volumes:
+        - /etc/antrea/:/etc/antrea
+        - /var/run/antrea:/var/run/antrea
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /var/lib/cni:/var/lib/cni
+        - /var/log/antrea:/var/log/antrea
+        - /proc:/host/proc
+        - /var/run/netns:/host/var/run/netns
+        - /run/xtables.lock:/run/xtables.lock
+    environment:
+        - NODE_NAME=$NODE_NAME
+    command: $ANTREA_AGENT_BIN --config $AGENT_CONF_PATH/antrea-agent.conf --logtostderr=false --log_file=$AGENT_LOG_DIR/antrea-agent.log
+    network_mode: host
+    privileged: true
+    cap_add:
+        - SYS_MODULE
+EOF
+}
+
+load_ovs_modules() {
+    for module in "${OVS_MODULES[@]}"; do
```

Comment:
OVS_MODULES is local to this function and not used anywhere else.. Dont make it global var